site/_posts/2022-10-30-spring-and-fusionAuth.md

```diff
@@ -0,0 +1,353 @@
+
+@Controller
+public class HomeController {
+
+  @Value("${spring.application.name}")
+  String appName;
+
+  @RequestMapping("/")
+  public String homePage(Model model) {
+      model.addAttribute("appName", appName);
+      return "home";
+  }  
+}
+```
+
+Under the `main/java/.../..../resources/templates` folder, create a new file named `home.html`. Add the following contents to the file:
+
+```html
+<html xmlns:th="http://www.w3.org/1999/xhtml" lang="en">
+<head><title>Home Page</title></head>
+<body>
+	<h1>Hello !</h1>
+	<p>Welcome to <span th:text="${appName}">Our App</span></p>
+
+	<p>You can view your profile <a href="/user">here</a></p>
+</body>
+</html>
+```
+
+## Adding the user page and controller
+
+We set the `user-info-uri` property in the `application.properties` file and add the `openid` and `profile` scopes to the `scope` property so that Spring can automatically query FusionAuth for the user's profile and make it available to the app via the [`AuthenticationPrincipal` annotation parameter](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/core/annotation/AuthenticationPrincipal.html). We can use this to build a user controller to show the logged-in user their profile on FusionAuth. You can customize the profile fields stored on FusionAuth and requested during registration. Read more about this [here](https://fusionauth.io/docs/v1/tech/guides/advanced-registration-forms).
+
+Under the `main/java/.../....` folder, create a new file named `UserController.java`. Add the following contents to this file:
+
+```java
+package com.fusionauth.javafusion;
+
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+public class UserController {
+
+  public UserController() {
+
+  } 
+
+  @RequestMapping("/user")
+  public String userPage(Model model, @AuthenticationPrincipal OidcUser principal) {
+    if (principal != null) {
+      model.addAttribute("profile", principal.getClaims());
+    }
+    return "user";
+  }
+}
+```
+
+Under the `main/java/.../..../resources/templates` folder, create a new file named `user.html`. Add the following contents to the file:
+
+```html
+<html xmlns:th="http://www.w3.org/1999/xhtml" lang="en">
+<head><title>User Profile</title></head>
+<body>
+	<h1>Welcome to the protected User page. Below is your OpenID profile information.</h1>
+	<p>Profile: <span th:text="${profile}"></span></p>
+
+	<h2>You can logout here: <a href="<YOUR_FUSIONAUTH_URL>/oauth2/logout?client_id=<YOUR_FUSIONAUTH_APP_CLIENT_ID>">Logout</a></h2>
+</body>
+</html>
+```
+
+Replace the values in the logout link:
+
+- `<YOUR_FUSIONAUTH_APP_CLIENT_SECRET>` with the client secret from the FusionAuth app created earlier.
```

Comment:
I don't see where to put this value in? Did you mean the client id?

Reply:
Yes, you are correct, it is meant to be the client ID. Thanks, updated